### PR TITLE
Fix Syntax Errors for Python 3

### PR DESCRIPTION
**Pull request by cclauss**
Fixes a bunch of __make lint__ issues around converting old style exceptions to new style for Python 3.

### CODE REVIEW

**Comment by tfmorris** [comment]
Fixing just the exceptions is already a pretty big PR, even without the miscellaneous white space changes, `while 1:` => `while True:`, etc changes. It's OK to create a PR with a bunch of small miscellaneous changes, if they're too small to justify their own PR, but if you've already got something weighty, like the exception changes, please try to resist the temptation to fold in more stuff.

I'll still put it on my list to review, but please keep this in mind for future PRs.

File: openlibrary/plugins/openlibrary/tests/test_home.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def test_about_template(self, render_template):
         posts = [web.storage({
             "title": "Blog-post-0",
             "link": "http://blog.openlibrary.org/2011/01/01/blog-post-0",
-            "pubdate": datetime.datetime(2011, 0o1, 0o1)
+            "pubdate": datetime.datetime(2011, 1, 1)
```
Comment (tfmorris) [inline openlibrary/plugins/openlibrary/tests/test_home.py]: While a little clearer, this is unrelated to Python 3

File: openlibrary/tests/core/test_processors_invalidation.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def test_hook(self, monkeypatch):
             "key": "/templates/site.tmpl",
             "type": "/type/template"
         }
-        web.ctx.site.save(doc, timestamp=datetime.datetime(2010, 0o1, 0o1))
+        web.ctx.site.save(doc, timestamp=datetime.datetime(2010, 1, 1))
```
Comment (tfmorris) [inline openlibrary/tests/core/test_processors_invalidation.py]: Another change not related to Python 3

Comment (cclauss) [reply openlibrary/tests/core/test_processors_invalidation.py]: $ __python3 -c "01"__  # --> SyntaxError: invalid token

File: openlibrary/plugins/openlibrary/tests/test_home.py (hunk 1 above)
Comment (cclauss) [reply openlibrary/plugins/openlibrary/tests/test_home.py]: $ __python3 -c "01"__  # --> SyntaxError: invalid token

Comment (cclauss) [reply openlibrary/plugins/openlibrary/tests/test_home.py]: Are we going to enter dates in octal?

Comment (cclauss) [reply openlibrary/plugins/openlibrary/tests/test_home.py]: https://github.com/internetarchive/openlibrary/pull/1466/files#diff-e9b7e7fd89beb0a8b81aad05951d6d38L37

Comment (cclauss) [reply openlibrary/plugins/openlibrary/tests/test_home.py]: https://github.com/internetarchive/openlibrary/pull/1466/files#diff-e9b7e7fd89beb0a8b81aad05951d6d38L36

**Comment by cclauss** [comment]
This PR should only be merged AFTER #1487 or similar.

File: openlibrary/plugins/search/code.py [hunk 3]
```diff
@@ -239,7 +239,7 @@ def strip_duplicates(seq, init=[]):
 
         # we have somehow gotten some queries for facet tokens with no
         # inverse.  remove these from the list.
-        ft_pairs = filter(lambda (a,b): b, ft_pairs)
+        ft_pairs = filter(lambda a_b: a_b[1], ft_pairs)
```
Comment (hornc) [inline openlibrary/plugins/search/code.py]: Wow, I wasn't aware of this change in python3, (clearly I spend too much time working with 2.7 :( )  thank you for alerting me to it. More info for anyone else interested: https://www.python.org/dev/peps/pep-3113/

@cclauss I agree with @tfmorris re. breaking this up into PRs that fix only one kind of issue (which I note you've done in #1487 :+1: ) Some change types will take less discussion / review than others and will get merged quicker.

For example, I'm not yet sure how I feel about the lambda signature changes, I guess I'll have to cope though as the PEP has been accepted ;)

File: openlibrary/plugins/openlibrary/tests/test_home.py (hunk 1 above)
Comment (hornc) [reply openlibrary/plugins/openlibrary/tests/test_home.py]: The original code cannot have been deliberate octal, it made me chuckle, because it looks sensible as a zero-padded date, but what is the sequence for other dates going to be?: `(01, 02, 03, 04, 05, 06, 07, 8, 9, 10)`?  or perhaps  `..., 07, 010, ...` ? Either a copy-paste or visual formatting error/choice that just happened to work out.

I agree `datetime.datetime(2011, 1, 1)` is the correct fix here, but I am enjoying the idea of octal dates :)   February has 28 days, but sometimes 035 when over-compensating... etc

**Comment by cclauss** [comment]
Is this PR now reviewable or should I make another subset?  If so, which changes would you recommend that I break out?

File: openlibrary/plugins/openlibrary/tests/test_home.py (hunk 1 above)
Comment (cclauss) [reply openlibrary/plugins/openlibrary/tests/test_home.py]: These changes were already merged in #1492

File: openlibrary/plugins/search/solr_client.py [hunk 4]
```diff
@@ -415,7 +417,7 @@ def facet_counts(result_list, facet_fields):
                 facets_k[x] += 1
 
     return filter(snd, ((f, sorted(facets[f].items(),
-                                   key=lambda (a,b): (-b,a)))
+                                   key=lambda a_b: (-a_b[1],a_b[0])))
```
[the remark this answers is not in the document]
Comment (cclauss) [reply openlibrary/plugins/search/solr_client.py]: __lambda (a, b): a__ is a syntax error in Python 3 but __lambda a, b: a__ is not.

**Comment by cclauss** [comment]
@tfmorris @hornc @mekarpeles

Can I please get your re-review on this one?  It now only deals with the Python keyword __raise__.

**Comment by tfmorris** [comment]
LGTM. All the comments made by @mekarpeles appear to have been addressed as well.
